packages/core/injector/module.ts

```diff
@@ -344,6 +345,9 @@ export class Module {
   }
 
   public isCustomValue(provider: any): provider is ValueProvider {
+    if (isObject(provider)) {
+      return provider.hasOwnProperty('useValue');
+    }
     return !isUndefined((provider as ValueProvider).useValue);
```

Comment:
```suggestion
    return isObject(provider) && provider.hasOwnProperty('useValue');
```
isn't the other check redundant now?